### PR TITLE
ci: when arch code is changed, run only on that arch

### PR DESCRIPTION
**Pull request by nashif**
In cases where only architecture code is changed, run on those architectures and nothing more. Speed up CI this way.

### CODE REVIEW

**Review by carlocaione — APPROVED** [review]

**Review by katsuster — APPROVED** [review]

File: scripts/ci/get_twister_opt.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                description="Generate twister argument files based on modified file")
+    parser.add_argument('-c', '--commits', default=None,
+            help="Commit range in the form: a..b")
+    return parser.parse_args()
+
+def find_archs(files):
+    archs = set()
+
+    for f in files:
+        if f.endswith(".rst") or f.endswith(".png") or f.endswith(".jpg"):
```
Comment (galak) [inline scripts/ci/get_twister_opt.py:32]: The filename filter seems like it should be pulled out of each `find_<FOO>`.

File: scripts/ci/get_twister_opt.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+                description="Generate twister argument files based on modified file")
+    parser.add_argument('-c', '--commits', default=None,
+            help="Commit range in the form: a..b")
+    return parser.parse_args()
+
+def find_archs(files):
```
Comment (galak) [inline scripts/ci/get_twister_opt.py:28]: Add a comment about how the matching of file to `arch` is happening.

**Review by galak — COMMENTED** [review]
Doesn't need to be done in this PR, but thinking about a follow up that is 'filename/path' to 'tag'.  For example kernel 'tagged' tests only if 'kernel/', 'arch/', ... are modified.

File: scripts/ci/get_twister_opt.py (hunk 1 above)
Comment (nashif) [reply scripts/ci/get_twister_opt.py:32]: why? We do not want to trigger builds when only those files with rst,png and jpg have changed.

**Review by galak — APPROVED** [review]

**Pull request merged**